src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -296,4 +296,52 @@ bool Series::checkLevelInterval(unsigned int cycleDuration, unsigned int optimiz
     return true;
 }
 
+bool Series::validateCycle(unsigned int firstHourOfTheWeek, std::optional<double> initialLevel,
+        unsigned int cycleDuration) const
+{
+    if (initialLevel.has_value())
+    {
+        for (unsigned int hour = firstHourOfTheWeek;
+                hour < firstHourOfTheWeek + Antares::Constants::nbHoursInAWeek;
+                hour += cycleDuration)
+        {
+            if (upperRuleCurve[hour] < initialLevel || lowerRuleCurve[hour] > initialLevel)
+            {
+                logs.warning() << "Error at line: " << hour + 1 << " for sts series upper or " <<
+                    "lower rule curves, initial level is not between those values";
+
+                return false;
+            }
+        }
+    }
+    else
+    {
+        double minBase = lowerRuleCurve[firstHourOfTheWeek];
+        double maxBase = upperRuleCurve[firstHourOfTheWeek];
+
+        for (unsigned int hour = firstHourOfTheWeek + cycleDuration;
+                hour < firstHourOfTheWeek + Antares::Constants::nbHoursInAWeek;
+                hour += cycleDuration)
+        {
+            double minCycle = lowerRuleCurve[hour];
+            double maxCycle = upperRuleCurve[hour];
+
+            if (maxCycle < minBase || minCycle > maxBase)
+            {
+                logs.warning() << "Error at line: " << hour + 1 << " for sts series upper"
+                    << " or lower rule curves, values at the start of the cycle are outside of" <<
+                    " those values";
+
+                return false;
+            }
+
+            // reduce the interval if necessary
+            minBase = std::max(minBase, minCycle);
+            maxBase = std::min(maxBase, maxCycle);
```

Comment:
You could simplify this algorithm a bit
1. Compute `b=`the min of all `upperRuleCurve[hour]` for all hours in the cycle set
2. Compute `a=`the max of all `lowerRuleCurve[hour]` for all hours in the cycle set
3. Check that `a < b` 

You could extract 1. and 2. into a `getBoundsForInitialLevel(uint firstHourOfTheWeek)` method, so that it can be re-used by the modeler.

Reply:
You can initialize the min with [`std::numeric_limits<double>::infinity()`](https://en.cppreference.com/w/cpp/types/numeric_limits/infinity) and the max with `-std::numeric_limits<double>::infinity()`